.github/workflows/build.yml

```diff
@@ -10,6 +10,16 @@ env:
   STACK_VERSION: '2.7.3'
 
 jobs:
+  static:
+    name: statically-linked exe / ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cachix/install-nix-action@v17
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - run: nix-build
```

Comment:
One potential problem with this is that there is a non-trivial amount of Nix code in `default.nix` to be able to get Spago to compile with Nix.  For instance, overrides to the `aeson` version, unbreaking `bower-json`, the workaround for the `docsSearchApp` thing, etc.  It would be unfortunate for a non-Nix user to send a PR to `spago`, have the Nix build break, and then not know how to fix it.

I'd of course be able to help with stuff like this, but I'm not sure a random contributor would know to ping me.